docs/api-docs/apps/tutorials/sample-app-next/06-step-4-enhance-ux.md

```diff
@@ -0,0 +1,797 @@
+
+```js
+export default async function list(req: NextApiRequest, res: NextApiResponse) {
+    try {
+        const { accessToken, storeHash } = await getSession(req);
+        const bigcommerce = bigcommerceClient(accessToken, storeHash);
+        // Optional: pass in API params here
+        const params = [
+            'limit=11',
+        ].join('&');
+
+        const { data } = await bigcommerce.get(`/catalog/products?${params}`);
+        res.status(200).json(data);
+    } catch (error) {
+        const { message, response } = error;
+        res.status(response?.status || 500).json({ message });
+    }
+}
+```
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-4-big-design/pages/api/products/list.ts)
+
+## Update custom hooks
+
+1. In the `lib` folder, open the `hooks.ts` file.
+
+2. Add the ` isLoading` property to the `useProducts` custom hook.
+
+```js
+export function useProducts() {
+    const encodedContext = useSession()?.context;
+    // Request is deduped and cached; Can be shared across components
+    const { data, error } = useSWR(encodedContext ? ['/api/products', encodedContext] : null, fetcher);
+
+    return {
+        summary: data,
+        isLoading: !data && !error,
+        isError: error,
+    };
+}
+```
+
+3. Add the `useProductList` custom hook.
+
+```js
+export function useProductList() {
+    const encodedContext = useSession()?.context;
+    // Use an array to send multiple arguments to fetcher
+    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext] : null, fetcher);
+
+    return {
+        list: data,
+        isLoading: !data && !error,
+        isError: error,
+        mutateList,
+    };
+}
+```
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-4-big-design/lib/hooks.ts)
+
+## Create the Products page
+
+1. In the `pages` folder, create a `products` folder.
+
+2. In the `products` folder, create an `index.tsx` file.
+
+3. At the top of the file, import the following packages:
+
+```js
+import { Button, Dropdown, Panel, Small, StatefulTable, Link as StyledLink } from '@bigcommerce/big-design';
+import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { ReactElement } from 'react';
+import { useProductList } from '../../lib/hooks';
+```
+<div class="HubBlock--callout">
+<div class="CalloutBlock--info">
+<div class="HubBlock-content">
+
+> ### Note
+> Because BigDesign and Next.js both have a component called `Link`, you will need to import BigDesign's `Link` as `StyledLink` to avoid TypeScript errors.
+
+</div>
+</div>
+</div>
+
+3. Add the `Products` functional component.
+
+```js
+const Products = () => {
+    const router = useRouter();
+    const { isError, isLoading, list = [] } = useProductList();
```

Comment:
again, we might want to add in some inline comments to explain to devs what we're doing

e.g.
```tsx
// get data from the v3 catalog/products endpoint
// https://developer.bigcommerce.com/api-reference/store-management/catalog/products/getproducts
const { isError, isLoading, list = [] } = useProductList(); 
// Properly format data for BigDesign's StatefulTable
const tableItems = list.map(({ id, inventory_level: stock, name, price }) => ({
    id,
    name,
    price,
    stock,
}));
```